docs/blueprints/EP025.rst

```diff
@@ -0,0 +1,111 @@
+
+- ``switches.<dpid>.metadata.ignored_loops``:
+
+This is for configuring which loops should be ignored on a ``<dpid>``, the value is a list of list pair of interface port numbers. For instance, if it's desirable to ignore a loop between interface number 5 and 6, and also between port 1 and itself: ``[[5, 6], [1,1]]``
+
+- ``interfaces.<interface_id>.metadata.looped``:
+
+This ``looped`` attribute is for state management, it will contain the following object whenever a loop is in place:
+
+.. code-block:: python3
+
+   {
+     'port_numbers': [<int>, <int>]
+     'detected_at': <str>, in this format %Y-%m-%dT%H:%M:%S
+   }
+
+Looped states should be removed if the interface has been disabled or deactivated, also there should be a periodic routine to remove the ``looped`` metadata interface state if the loop hasn't been detected anymore, this clean up routine should run at a lower pace than of_lldp polling, usually 3-4 times slower, and if the loop hasn't been seen within this time frame again then it would assume that the loop has stopped.
+
+The following options would be configurable on settings:
+
+- ``LLDP_LOOP_ACTIONS``:
+
+This is for configuring which actions should be taken once a loop is found. The actions are a list of strings, ``log`` and ``disable`` are both supported. The ``log`` action will log an warning message informing that a loop was detected on a dpid and on which interfaces. The ``disable`` action will disable an interface via the ``topology`` NApp, and eventually enable it again if the loop stops. For instance, ``['log', 'disable']`` would apply both actions to any loop that isn't ignored and is found. The default actions would be ``['log']`` based on what's expected to be used in production in this first iteration.
+
+- ``LOOP_LOG_EVERY``:
+
+This is for configuring which frequency in seconds the `log` action should emit a log whenever a loop is found, by default it's 60 seconds, this is for avoiding a flood of messages.
+
+
+Events
+======
+
+The following events will be published and subscribed.
+
+Subscribed
+----------
+
+- ``kytos/topology.topology_loaded``
+- ``kytos/topology.link_down``
+- ``kytos/topology.interface.disabled`` (not implemented on topology yet)
+
+Published
+---------
+
+kytos.of_lldp/loop.detected
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*buffer*: ``app``
+
+Event reporting that a loop has been detected.
+
+Content:
+
+.. code-block:: python3
+
+   {
+     'dpid': <str>,
+     'interface_id': <str>,
+     'port_numbers': [<int>, <int>],
+   }
+
+
+kytos.of_lldp/loop.stopped
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*buffer*: ``app``
+
+Event reporting that a loop has been stopped.
+
+Content:
+
+.. code-block:: python3
+
+   {
+     'dpid': <str>,
+     'interface_id': <str>,
+     'port_numbers': [<int>, <int>],
+   }
```

Comment:
Sure, let's map what's going to be needed. 

1 - Endpoint to return all the current detected loops, maybe we can return this structure here, indexed by each dpid, and each primary looped interface, that's close enough to how we have it stored, let me know if that would suffice:

```
{
  "00:00:00:00:00:00:00:01": {
    "00:00:00:00:00:00:00:01:1": {
      "state": "detected",
      "port_numbers": [
        1,
        2
      ],
      "updated_at": "2022-03-02T18:55:51",
      "detected_at": "2022-03-02T18:55:26"
    }
  }
}
```

2  - I think we can "force" checking (or reading again) the existing detected loops when consulting the endpoint that list all of the existing loops (and if any new were detected in the meantime it would be there), that would it simplify or did you mean a way to force to send new LLDP probes to check for loops again?

Reply:
I've synced with Jeronimo, we'll address this in a next iteration.